File: client-app/src/schemas/sharedValidators/emailValidator.ts
import { z } from 'zod'

export function emailValidator(messageIdentifier?: string) {
  const Email = messageIdentifier ?? 'Email'

  return z.string()
    .min(1, { message: `${Email} obrigatório` })
    .email(`Formato de ${Email} inválido`)
}

File: Application/Validators/SharingValidators/EmailValidator.cs
using FluentValidation;

namespace ICorteApi.Application.Validators;

internal static class EmailValidationExtensions
{
    public static IRuleBuilderOptions<T, string> ApplyEmailValidation<T>(this IRuleBuilder<T, string> ruleBuilder)
    {
        return ruleBuilder
            .NotEmpty().WithMessage("Email obrigatório")
            .DependentRules(() =>
            {
                ruleBuilder
                    .EmailAddress().WithMessage("Formato de email inválido");
            });
    }
}
